dap_aria_mapping/utils/validation.py

```diff
@@ -0,0 +1,568 @@
+        if len(clust_dict.keys()) == 0:
+            cluster_journal_rank[clust_label] = {}
+            continue
+        for entity, ecount in clust_dict.items():
+            val = entity_journal_prop[entity]
+            for journal, jprop in val.items():
+                jcount = cluster_journal_counts[clust_label][journal]
+                if journal in cluster_journal_rank[clust_label].keys():
+                    cluster_journal_rank[clust_label][journal] += (
+                        np.log(2 + (ecount / nconst)) ** 2
+                        * np.log(1 + (jcount / nconst)) ** 2
+                        * jprop
+                    )
+                else:
+                    cluster_journal_rank[clust_label][journal] = (
+                        np.log(2 + (ecount / nconst)) ** 2
+                        * np.log(1 + (jcount / nconst)) ** 2
+                        * jprop
+                    )
+    return cluster_journal_rank
+
+
+def get_cluster_journal_labels(
+    cluster_journal_ranks: Dict[str, Dict[str, int]], num_labels: int = 3
+) -> Dict[str, str]:
+    """Creates a dictionary of journal labels for each cluster.
+
+    Args:
+        num_labels (int, optional): The number of labels to return.
+            Defaults to 3.
+
+    Returns:
+        Dict[str, str]: A dictionary of journal labels for each cluster.
+
+            {
+                cluster_label: "Journals: journal, journal, journal",
+                ...
+            }
+    """
+    cluster_tuples = {}
+    for k, v in cluster_journal_ranks.items():
+        cluster_tuples[k] = sorted(
+            [tuple([entity, count]) for entity, count in v.items()],
+            key=lambda x: x[1],
+            reverse=True,
+        )
+
+    return {
+        k: "Journals: " + ", ".join([x[0] for x in v[:num_labels]])
+        for k, v in cluster_tuples.items()
+    }
+
+
+def get_main_entity_journal(
+    entity: str, entity_journal_counts: Dict[str, Dict[str, int]]
+) -> str:
+    """Returns the journal with the most entities for a given entity.
+
+    Args:
+        entity (str): The entity to get the journal for.
+        entity_journal_counts (Dict[str, Dict[str, int]]): A dictionary
+            of entity counts in journals.
+
+    Returns:
+        str: The journal with the most entities for a given entity.
+    """
+    journal_counts = entity_journal_counts[entity]
+    if len(journal_counts.keys()) == 0:
+        return "Other"
+    max_count = max(journal_counts.values())
+    max_journals = [k for k, v in journal_counts.items() if v == max_count]
+    return random.choice(max_journals)
+
+
+def build_labelled_taxonomy(
+    df: pd.DataFrame,
+    journal_entities: Dict[str, Sequence[str]],
+    entity_counts: Dict[str, int],
+    entity_journal_counts: Dict[str, Dict[str, int]],
+    levels: int = 3,
+) -> pd.DataFrame:
+    """Builds a labelled taxonomy from a taxonomy dataframe, a
```

Comment:
Would be helpful to be a bit more descriptive about what the 'labeled taxonomy' is

Reply:
Hopefully clearer now.